src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Common/SyntaxNodeExtensions.cs

```diff
@@ -0,0 +1,212 @@
+                .OfType<T>()
+                .FirstOrDefault(m =>
+                    (m is CSSyntax.MethodDeclarationSyntax csM &&
+                    csM.Identifier.ToString().Equals(methodName, StringComparison.Ordinal) &&
+                    requiredParameterTypes.All(req => csM.ParameterList.Parameters.Any(p => string.Equals(p.Type?.ToString(), req, StringComparison.Ordinal))))
+                || (m is VBSyntax.MethodStatementSyntax vbM &&
+                    vbM.Identifier.ToString().Equals(methodName, StringComparison.Ordinal) &&
+                    requiredParameterTypes.All(req => vbM.ParameterList.Parameters.Any(p => string.Equals(p.AsClause?.Type?.ToString(), req, StringComparison.Ordinal)))));
+
+        /// <summary>
+        /// Applies whitespace trivia and new line trivia from another syntax node to this one.
+        /// </summary>
+        /// <typeparam name="T">The type of syntax node to be updated.</typeparam>
+        /// <param name="statement">The syntax node to update.</param>
+        /// <param name="otherStatement">The syntax node to copy trivia from.</param>
+        /// <returns>The original syntax node updated with the other syntax's whitespace and new line trivia.</returns>
+        public static T WithWhitespaceTriviaFrom<T>(this T statement, SyntaxNode otherStatement)
+            where T : SyntaxNode
+        {
+            return statement
+                .WithLeadingTrivia(otherStatement?.GetLeadingTrivia().Where(IsWhitespaceTrivia) ?? SyntaxTriviaList.Empty)
+                .WithTrailingTrivia(otherStatement?.GetTrailingTrivia().Where(IsWhitespaceTrivia) ?? SyntaxTriviaList.Empty);
+
+            static bool IsWhitespaceTrivia(SyntaxTrivia trivia) =>
+                CSharpExtensions.IsKind(trivia, CS.SyntaxKind.EndOfLineTrivia)
+                || CSharpExtensions.IsKind(trivia, CS.SyntaxKind.WhitespaceTrivia)
+                || VisualBasicExtensions.IsKind(trivia, VB.SyntaxKind.EndOfLineTrivia)
+                || VisualBasicExtensions.IsKind(trivia, VB.SyntaxKind.WhitespaceTrivia);
+        }
+
+        /// <summary>
+        /// Determines if a syntax node includes a using or import statement for a given namespace.
+        /// </summary>
+        /// <param name="node">The node to analyze.</param>
+        /// <param name="namespaceName">The namespace name to check for.</param>
+        /// <returns>True if the node or any of its descendents have a an import or using statement for the given namespace. False otherwise.</returns>
+        public static bool HasUsingStatement(this SyntaxNode node, string namespaceName)
+        {
+            if (node is null)
+            {
+                throw new ArgumentNullException(nameof(node));
+            }
+
+            var nodes = node.DescendantNodesAndSelf(_ => true, false);
+
+            return nodes.OfType<CSSyntax.UsingDirectiveSyntax>().Any(u => u.Name.ToString().Equals(namespaceName, StringComparison.Ordinal))
```

Comment:
This doesn't seem to track if it would be in scope. Maybe not a big deal, but it would be good to call that out as a corner case we're not worried about

Reply:
It doesn't. That's probably why its behavior is different from the simplifier (though, the simplifier apparently doesn't check scope either, it just defaults differently than we do - it assumes not in scope while this assumes in-scope).

Originally, I left it out as a low-pri corner case, but it's probably not too hard to do right. I'll update the method to at least make an effort to consider scope.